prover/zkevm/arithmetization/arithmetization.go

```diff
@@ -41,19 +56,61 @@ func NewArithmetization(builder *wizard.Builder, settings Settings) *Arithmetiza
 	return &Arithmetization{
 		Schema:   schema,
 		Settings: &settings,
+		Metadata: metadata,
 	}
 }
 
 // Assign the arithmetization related columns. Namely, it will open the file
 // specified in the witness object, call corset and assign the prover runtime
-// columns.
+// columns. As part of the assignment processs, the original trace is expanded
+// according to the given schema.  The expansion process is about filling in
+// computed columns with concrete values, such for determining multiplicative
+// inverses, etc.
 func (a *Arithmetization) Assign(run *wizard.ProverRuntime, traceFile string) {
-
 	traceF := files.MustRead(traceFile)
-	trace, errT := ReadLtTraces(traceF, a.Schema)
+	// Parse trace file and extract raw column data.
+	rawColumns, metadata, errT := ReadLtTraces(traceF, a.Schema)
+
+	// Performs a compatibility check by comparing the constraints
+	// commit of zkevm.bin with the constraints commit of the trace file.
+	// Panics if an incompatibility is detected.
+	if *a.Settings.IgnoreCompatibilityCheck == false {
```

Comment:
@DavePearce I moved this to the top of Assign() so that constraints incompatibility fails earlier than other errors  (e.g., error loading the trace, fpath=%q, err=%v).